coretex/configuration/base.py

```diff
@@ -0,0 +1,79 @@
+#     Copyright (C) 2023  Coretex LLC
+
+#     This file is part of Coretex.ai
+
+#     This program is free software: you can redistribute it and/or modify
+#     it under the terms of the GNU Affero General Public License as
+#     published by the Free Software Foundation, either version 3 of the
+#     License, or (at your option) any later version.
+
+#     This program is distributed in the hope that it will be useful,
+#     but WITHOUT ANY WARRANTY; without even the implied warranty of
+#     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#     GNU Affero General Public License for more details.
+
+#     You should have received a copy of the GNU Affero General Public License
+#     along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+from typing import Dict, Any, Optional, Type, TypeVar
+from abc import abstractmethod
+from pathlib import Path
+
+import os
+import json
+
+
+T = TypeVar("T", int, float, str, bool)
+
+CONFIG_DIR = Path.home().joinpath(".config", "coretex")
+DEFAULT_VENV_PATH = CONFIG_DIR / "venv"
+
+
+class InvalidConfiguration(Exception):
+    pass
+
+class BaseConfiguration:
+
+    def __init__(self, path: Path) -> None:
```

Comment:
Let's change the constructor to accept a loaded configuration dictionary (`raw`). Add a new class function `load`. Add `isConfigured` and `isConfigValid` functions to `BaseConfiguration`. When `load` is called if config file does not exist of `isConfigured` returns `False` raise `ConfigurationNotFound` exception, and if `isConfigValid` returns `False` raise `InvalidConfiguration` exception. Based on these 2 exceptions prompt for a completely new configuration, or updating the existing one.